### PR TITLE
misc.cache.memoize_method: expose cache_clear() on memoized method

### PR DESCRIPTION
**Pull request by kernc**
##### Issue
`.cache_clear()` / `.cache_info()` were not exposed on methods memoized with `misc.cache.memoize_method()`.

##### Description of changes
Expose by refactoring. Hope it's correct.

##### Includes
- [X] Code changes
- [X] Tests
- [ ] Documentation

### CODE REVIEW

File: Orange/misc/cache.py [hunk 1]
```diff
@@ -41,20 +41,16 @@ def memoize_method(*lru_args, **lru_kwargs):
...
 
+        @lru_cache(*lru_args, **lru_kwargs)
+        def _cached_method(self_weak, *args, **kwargs):
+            return func(self_weak(), *args, **kwargs)
+
         @wraps(func)
         def _wrapped_func(self, *args, **kwargs):
-            self_weak = weakref.ref(self)
-            # We're storing the wrapped method inside the instance. If we had
-            # a strong reference to self the instance would never die.
```
Comment (janezd) [inline Orange/misc/cache.py]: Please explain in the docstring why is this needed and how is it different from lru_cache.

**Comment by astaric** [comment]
I have updated the docstring. Once the checks pass this can be merged.
